featureswitch/featureswitch.go

```diff
@@ -0,0 +1,55 @@
+package featureswitch
+
+import (
+	"crypto/md5"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+
+	"github.com/mongodb/grip"
+)
+
+const s3WriteFeatureSwitch = "LK_WRITE_S3_FEATURE_SWITCH"
+
+func hashToFloat(hash []byte) float64 {
+	value := binary.BigEndian.Uint32(hash[:4])
+	// The plus one ensures that we return a number in the range 0 to 1,
+	// inclusive of 0 and exclusive of 1.
+	return float64(value) / (math.MaxUint32 + 1)
```

Comment:
Is this reliant on running on the size of an `int` being larger than 32 bits so it doesn't overflow?

Reply:
It's an untyped numeric constant so it doesn't overflow. This is also captured in the tests.

Reply:
Basically what I _think_ happens is that it's converted to a float64 for the purposes of the division, and should be exactly representable, since it is only 33 bits.

Reply:
How do the tests prove this out?

Reply:
I convert it to a float64 to ensure that it uses float division and not integer division (which would always give us 1 / 0).

The tests demonstrate that given 0xFFFFFFFF (max uint) we output a number very close to 1, as opposed to overflowing in some fashion and giving an odd result (+Inf seems like the most likely, since overflowing uint should go to 0 and floating point division by zero should return +Inf).

Reply:
That overflow happens because it needs to store in variable, and it actually happens at compile time from what I can tell, since Go can tell that the constant value exceeds what it can store in an int64, which I'm guessing is the largest int type it will consider?

I an also make sure that we cast the denominator immediately to a float64 if that seems less iffy.

Reply:
Got it.
Either way is good by me